### DISCUSSION TITLE
Websocket fragmenting in Axum

### DISCUSSION THREAD

**Discussion by uwueviee** [tokio-rs/axum#2024] [Q&A] [answered]
Due to a restriction of the client I am building the server for, I need to be able to send fragmented binary messages down the websocket. As far as I can tell, I am unable to do this due to not having the ability to create my own frames. Is there another way of doing websocket fragmentation within Axum's websockets or a way to get the original tungstenite websocket where I'd be able to make my own frames?

**Comment by davidpdrsn** [comment, marked as answer]
Try using https://crates.io/crates/axum-tungstenite instead. That integrates directly with tungstenite and uses their types directly. So you can create your own frames.

**Reply by uwueviee** [reply]
Oh huh wasn't aware of that create, thanks! I'll look into that
